src/ripple/ledger/impl/View.cpp

```diff
@@ -391,7 +391,11 @@ transferRate (ReadView const& view,
     auto const sle = view.read(keylet::account(issuer));
 
     if (sle && sle->isFieldPresent (sfTransferRate))
+    {
+        if (sfTransferRate > 2 * QUALITY_ONE)
+            return Rate{ 2 * QUALITY_ONE };
         return Rate{ sle->getFieldU32 (sfTransferRate) };
+    }
```

Comment:
So, you could collapse all this to something like:

```return Rate{ std::min((*sle)[sfTransferRate], 2 * QUALITY_ONE) };```

But since we want this code to be on an amendment, we'll need something more verbose:

Consider:
```auto const rate = (*sle)[sfTransferRate];
if (/*amendmentActive*/ && rate > maxRate)
    rate = std::min(rate, maxRate);
return Rate{rate};
```